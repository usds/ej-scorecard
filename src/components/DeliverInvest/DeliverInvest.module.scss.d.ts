declare namespace DeliverInvestNamespace {
  export interface IDeliverInvestScss {
    h2: string;
    summaryBox: string;
  }
}

declare const DeliverInvestScssModule: DeliverInvestNamespace.IDeliverInvestScss & {
  /** WARNING: Only available when "css-loader" is used without "style-loader" or "mini-css-extract-plugin" */
  locals: DeliverInvestNamespace.IDeliverInvestScss;
};

export = DeliverInvestScssModule;
