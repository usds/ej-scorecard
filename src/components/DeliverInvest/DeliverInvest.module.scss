@use "uswds-core" as *;

.h2 { 
    @include u-margin-top(4);
    border-bottom: 1px solid #e8e8f0;
}

.summaryBox {
    @include u-margin-top(10);
}