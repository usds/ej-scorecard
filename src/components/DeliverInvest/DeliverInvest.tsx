import React from 'react';
import {
  Grid,
  SummaryBox,
  SummaryBoxContent,
  SummaryBoxHeading,
  Table,
} from '@trussworks/react-uswds';
import * as styles from './DeliverInvest.module.scss';
import { DeliverInvestProps } from '@/types';

const DeliverInvest: React.FC<DeliverInvestProps> = ({ deliverInvest }) => {
  const table1Data = (
    <>
      <thead>
        <tr>
          <th colSpan={4}>{deliverInvest.node.T1_Title}</th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <th scope="row" colSpan={2}>
            {deliverInvest.node.Cat1}
          </th>
          <td className="font-mono-sm text-tabular text-center">
            {deliverInvest.node.Cat1_N1}
          </td>
          <td className="font-mono-sm text-tabular text-center">
            {deliverInvest.node.Cat1_N2}
          </td>
        </tr>
        <tr>
          <th scope="row" colSpan={2}>
            {deliverInvest.node.Cat2}
          </th>
          <td className="font-mono-sm text-tabular text-center">
            {deliverInvest.node.Cat2_N1}
          </td>
          <td className="font-mono-sm text-tabular text-center">
            {deliverInvest.node.Cat2_N2}
          </td>
        </tr>
        <tr>
          <th scope="row" colSpan={2}>
            {deliverInvest.node.Cat3}
          </th>
          <td className="font-mono-sm text-tabular text-center">
            {deliverInvest.node.Cat3_N1}
          </td>
          <td className="font-mono-sm text-tabular text-center">
            {deliverInvest.node.Cat3_N2}
          </td>
        </tr>
      </tbody>
    </>
  );
  const table2Data = (
    <>
      <thead>
        <tr>
          <th colSpan={4}>{deliverInvest.node.T2_Title}</th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <th scope="row" colSpan={2}>
            {deliverInvest.node.T2_M1}
          </th>
          <td className="font-mono-sm text-tabular text-center">
            {deliverInvest.node.T2_N1}
          </td>
        </tr>
      </tbody>
    </>
  );
  const table3Data = (
    <>
      <thead>
        <tr>
          <th colSpan={4}>{deliverInvest.node.T3_Title}</th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <th scope="row" colSpan={2}>
            {deliverInvest.node.T3_M1}
          </th>
          <td className="font-mono-sm text-tabular text-center">
            {deliverInvest.node.T3_N1}
          </td>
        </tr>
        <tr>
          <th scope="row" colSpan={2}>
            {deliverInvest.node.T3_M2}
          </th>
          <td className="font-mono-sm text-tabular text-center">
            {deliverInvest.node.T3_N2}
          </td>
        </tr>
        <tr>
          <th scope="row" colSpan={2}>
            {deliverInvest.node.T3_M3}
          </th>
          <td className="font-mono-sm text-tabular text-center">
            {deliverInvest.node.T3_N3}
          </td>
        </tr>
      </tbody>
    </>
  );
  const table4Data = (
    <>
      <thead>
        <tr>
          <th colSpan={4}>{deliverInvest.node.T4_Title}</th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <th scope="row" colSpan={2}>
            {deliverInvest.node.T4_M1}
          </th>
          <td className="font-mono-sm text-tabular text-center">
            {deliverInvest.node.T4_V1}
          </td>
        </tr>
      </tbody>
    </>
  );

  return (
    <>
      <Grid row>
        <Grid desktop={{ col: 8 }} tablet={{ col: 10 }} col={12}>
          <h2 className={styles.h2}>Delivering investments</h2>
        </Grid>
      </Grid>
      <Grid row>
        <Grid desktop={{ col: 6 }} tablet={{ col: 10 }} col={12}>
          Short narrative about the why delivering investments is important. I
          prem ins _dolor sit amet concertetur adinissing plitvam at magna vitan
          prat convall bandit. Phasellus id urna ultrices, vestibulum guam a.
          accumsan mauris
        </Grid>
      </Grid>
      <Grid row gap={6}>
        <Grid desktop={{ col: 8 }} tablet={{ col: 10 }} col={12}>
          <Table fullWidth bordered striped>
            {table1Data}
          </Table>
          <Table fullWidth bordered striped>
            {table2Data}
          </Table>
          <Table fullWidth bordered striped>
            {table3Data}
          </Table>
          <Table fullWidth bordered striped>
            {table4Data}
          </Table>
        </Grid>
        <Grid desktop={{ col: 4 }} tablet={{ col: 10 }} col={12}>
          <SummaryBox className={`margin-top-2`}>
            <SummaryBoxHeading headingLevel="h3">
              {`Delivering investments at ${deliverInvest.node.Agency_Name}`}
            </SummaryBoxHeading>
            <SummaryBoxContent>
              {deliverInvest.node.DeliverInvest}
            </SummaryBoxContent>
          </SummaryBox>
        </Grid>
      </Grid>
    </>
  );
};

export default DeliverInvest;
