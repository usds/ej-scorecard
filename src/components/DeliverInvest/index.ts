import DeliverInvest from './DeliverInvest';
export default DeliverInvest;
