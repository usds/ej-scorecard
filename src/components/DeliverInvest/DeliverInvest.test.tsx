import createComponentWithIntl from '@/test/testHelpers';

import DeliverInvest from './DeliverInvest';

const di = {
  node: {
    id: `8ab5f568-6353-5a43-b2c0-b31b78a74be5`,
    __typename: `DeliverInvestCsv`,
    Agency_Name: `Agency A`,
    DeliverInvest: `LoremA ipsum dolor sit amet, consectetur adipiscing elit. Morbi luctus velit non justo tempus, sit amet bibendum tellus convallis. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae; Nam in gravida mi. Nam nisl ante, ultricies et massa id, euismod sodales arcu. Sed velit felis, tristique a nunc sit amet, rutrum iaculis lectus. Nullam auctor justo at odio sagittis rutrum. Nunc eget malesuada mauris. Donec tincidunt ex in mi ultrices sodales. Aliquam at lobortis eros, eu mollis tellus. Nullam sit amet elementum tellus. Suspendisse viverra ligula neque, id dapibus nisl elementum et. Integer mattis magna ac lorem fermentum, laoreet accumsan tellus tincidunt. Donec bibendum viverra velit vel eleifend.`,
    T1_Title: `Table1 Metrics A`,
    Cat1: `Category 1A`,
    Cat1_N1: `#`,
    Cat1_N2: `$#`,
    Cat2: `Category 2A`,
    Cat2_N1: `#`,
    Cat2_N2: `$#`,
    Cat3: `Category 3A`,
    Cat3_N1: `#`,
    Cat3_N2: `$#`,
    T2_Title: `Table2 Metrics A`,
    T2_M1: `MetricA`,
    T2_N1: `$#`,
    T3_Title: `Table3 Metrics A`,
    T3_M1: `MetricA`,
    T3_N1: `$#`,
    T3_M2: `MetricA`,
    T3_N2: `$#`,
    T3_M3: `MetricA`,
    T3_N3: `$#`,
    T4_Title: `Table4 Metrics A`,
    T4_M1: `Optional: MetricA`,
    T4_V1: `LoremA ipsum dolor sit amet, consectetur adipiscing elit. Morbi luctus velit non justo tempus, sit amet bibendum tellus convallis. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae; Nam in gravida mi. Nam nisl ante, ultricies et massa id, euismod sodales arcu. Sed velit felis, tristique a nunc sit amet, rutrum iaculis lectus. Nullam auctor justo at odio sagittis rutrum. Nunc eget malesuada mauris. Donec tincidunt ex in mi ultrices sodales. Aliquam at lobortis eros, eu mollis tellus. Nullam sit amet elementum tellus. Suspendisse viverra ligula neque, id dapibus nisl elementum et. Integer mattis magna ac lorem fermentum, laoreet accumsan tellus tincidunt. Donec bibendum viverra velit vel eleifend.`,
  },
};

describe(`rendering of DeliverInvest Component`, () => {
  it(`checks if component renders`, () => {
    const component = createComponentWithIntl(
      <DeliverInvest deliverInvest={di} />,
    );

    // const tree = ;

    expect(component.toJSON()).toMatchSnapshot();
  });
});
