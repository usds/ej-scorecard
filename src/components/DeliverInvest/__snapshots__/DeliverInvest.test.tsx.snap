// Vitest Snapshot v1

exports[`rendering of DeliverInvest Component > checks if component renders 1`] = `
<div
  className="grid-row grid-gap-6"
  data-testid="grid"
>
  <div
    className="grid-col-12 tablet:grid-col-10 desktop:grid-col-8"
    data-testid="grid"
  >
    <h2>
      Delivering investments
    </h2>
    <table
      className="usa-table src-components-Table-Table-module__fullwidth--fOyj6 usa-table--striped"
      data-testid="table"
    >
      <thead>
        <tr>
          <th
            colSpan={4}
          >
            Table1 Metrics A
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <th
            colSpan={2}
            scope="row"
          >
            Category 1A
          </th>
          <td
            className="font-mono-sm text-tabular text-center"
          >
            #
          </td>
          <td
            className="font-mono-sm text-tabular text-center"
          >
            $#
          </td>
        </tr>
        <tr>
          <th
            colSpan={2}
            scope="row"
          >
            Category 2A
          </th>
          <td
            className="font-mono-sm text-tabular text-center"
          >
            #
          </td>
          <td
            className="font-mono-sm text-tabular text-center"
          >
            $#
          </td>
        </tr>
        <tr>
          <th
            colSpan={2}
            scope="row"
          >
            Category 3A
          </th>
          <td
            className="font-mono-sm text-tabular text-center"
          >
            #
          </td>
          <td
            className="font-mono-sm text-tabular text-center"
          >
            $#
          </td>
        </tr>
      </tbody>
    </table>
    <table
      className="usa-table src-components-Table-Table-module__fullwidth--fOyj6 usa-table--striped"
      data-testid="table"
    >
      <thead>
        <tr>
          <th
            colSpan={4}
          >
            Table2 Metrics A
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <th
            colSpan={2}
            scope="row"
          >
            MetricA
          </th>
          <td
            className="font-mono-sm text-tabular text-center"
          >
            $#
          </td>
        </tr>
      </tbody>
    </table>
    <table
      className="usa-table src-components-Table-Table-module__fullwidth--fOyj6 usa-table--striped"
      data-testid="table"
    >
      <thead>
        <tr>
          <th
            colSpan={4}
          >
            Table3 Metrics A
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <th
            colSpan={2}
            scope="row"
          >
            MetricA
          </th>
          <td
            className="font-mono-sm text-tabular text-center"
          >
            $#
          </td>
        </tr>
        <tr>
          <th
            colSpan={2}
            scope="row"
          >
            MetricA
          </th>
          <td
            className="font-mono-sm text-tabular text-center"
          >
            $#
          </td>
        </tr>
        <tr>
          <th
            colSpan={2}
            scope="row"
          >
            MetricA
          </th>
          <td
            className="font-mono-sm text-tabular text-center"
          >
            $#
          </td>
        </tr>
      </tbody>
    </table>
    <table
      className="usa-table src-components-Table-Table-module__fullwidth--fOyj6 usa-table--striped"
      data-testid="table"
    >
      <thead>
        <tr>
          <th
            colSpan={4}
          >
            Table4 Metrics A
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <th
            colSpan={2}
            scope="row"
          >
            Optional: MetricA
          </th>
          <td
            className="font-mono-sm text-tabular text-center"
          >
            LoremA ipsum dolor sit amet, consectetur adipiscing elit. Morbi luctus velit non justo tempus, sit amet bibendum tellus convallis. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae; Nam in gravida mi. Nam nisl ante, ultricies et massa id, euismod sodales arcu. Sed velit felis, tristique a nunc sit amet, rutrum iaculis lectus. Nullam auctor justo at odio sagittis rutrum. Nunc eget malesuada mauris. Donec tincidunt ex in mi ultrices sodales. Aliquam at lobortis eros, eu mollis tellus. Nullam sit amet elementum tellus. Suspendisse viverra ligula neque, id dapibus nisl elementum et. Integer mattis magna ac lorem fermentum, laoreet accumsan tellus tincidunt. Donec bibendum viverra velit vel eleifend.
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    className="grid-col-12 tablet:grid-col-10 desktop:grid-col-4"
    data-testid="grid"
  >
    <div
      className="usa-summary-box"
      data-testid="summary-box"
    >
      <div
        className="usa-summary-box__body"
      >
        <h3
          className="usa-summary-box__heading"
        >
          Delivering investments at Agency A
        </h3>
        <div
          className="usa-summary-box__text"
        >
          LoremA ipsum dolor sit amet, consectetur adipiscing elit. Morbi luctus velit non justo tempus, sit amet bibendum tellus convallis. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae; Nam in gravida mi. Nam nisl ante, ultricies et massa id, euismod sodales arcu. Sed velit felis, tristique a nunc sit amet, rutrum iaculis lectus. Nullam auctor justo at odio sagittis rutrum. Nunc eget malesuada mauris. Donec tincidunt ex in mi ultrices sodales. Aliquam at lobortis eros, eu mollis tellus. Nullam sit amet elementum tellus. Suspendisse viverra ligula neque, id dapibus nisl elementum et. Integer mattis magna ac lorem fermentum, laoreet accumsan tellus tincidunt. Donec bibendum viverra velit vel eleifend.
        </div>
      </div>
    </div>
  </div>
</div>
`;
